### ISSUE TITLE
Migration to aioredis 2.0

### ISSUE THREAD

**Issue by Andrew-Chen-Wang** [aio-libs/aioredis-py#930]
Hi all,

After #891, aioredis will be making a major jump to `aioredis==2.0.0`. The PR made aioredis an enormous async port of redis-py. For the migration docs, please refer to: https://github.com/aio-libs/aioredis-py/blob/master/docs/migration.md. To install:

```shell
pip install aioredis==2.0.0
```

<details><summary>Old pip install instructions for beta left for posterity</summary>

`pip install aioredis==2.0.0b1`

To install aioredis at a specific commit:

`$ pip install git+git://github.com/aio-libs/aioredis-py.git@ eb4c001a598c01d9a0e5f2a24065c486c823d1b0 `

To install at master:

`pip install git+git://github.com/aio-libs/aioredis-py.git@master`

</details>

We'd love to hear how well this new version of aioredis integrates with your current production code! @abrookins and I will be updating this specific GitHub issue with any updates that you should be aware of; any breakages can be reported as a separate GitHub issue or in this one, too.

Thanks everyone!

**Comment by bmoscon** [comment]
I've been using this for a week or two (just running off a commit hash from that time period). It would be nice to be able to push my project up to pypi, but I can't because pypi wont let you use a direct dependency. Have you considered releasing a 2.0.0a preview version (or something like that)?

**Comment by Andrew-Chen-Wang** [comment]
Yes we've considered it, but I don't have release permission only seandstewart, but @abrookins might? @abrookins I suggest we push out the alpha version. It won't show up as an upgrade to people but it at least makes the installation easier to do.

**Comment by bmoscon** [comment]
You can see the list of those with access here, there are only 3 

https://pypi.org/project/aioredis/

**Comment by bmoscon** [comment]
Hey @asvetlov - you're on the pypi account, any chance you could give access to the two Andrews so we can continue release work while the Sean is AFK? He made them admins on the repo, I doubt he'd be opposed to having them have pypi access

**Comment by seandstewart** [comment]
@bmoscon - 

thanks for stepping in with your vote of confidence. I’ve been afk for the last ten days because my father passed away and I’m helping my mother with everything. I commented on the Maintainership ticket letting folks know. 

https://github.com/aio-libs/aioredis-py/issues/822#issuecomment-808465684

@Andrew-Chen-Wang the GitHub action for automatic deployment from a tag is live in master so should only require a tagged commit on master - note I haven’t tested the action so it may require tweaking to get it right. If you’re unable to publish an alpha package via the action, let me know and I’ll step online to to get one published to pypi.

**Comment by bmoscon** [comment]
@seandstewart wow sorry to hear that! Take care of yourself/your family!

**Comment by abrookins** [comment]
I'll be more available to help out in May, but in the meantime I agree @Andrew-Chen-Wang, let's push an alpha 2.0 release. Pip should prevent people from installing an alpha unless they ask for it IIRC. Sounds like we need to tag 2.0.0a1 but also change the version in master to use the alpha signifier. I can work on that.

**Comment by abrookins** [comment]
@Andrew-Chen-Wang https://github.com/aio-libs/aioredis-py/pull/934

**Comment by bmoscon** [comment]
alas, it looks like its hung. Hopefully there is an easy way for it to retry. . . .

**Comment by abrookins** [comment]
Our first 2.0.0 alpha release is now available on PyPI: https://pypi.org/project/aioredis/2.0.0a1/

**Comment by bmoscon** [comment]
I'm already using it :D

**Comment by rafaelSorel** [comment]
Hi guys, is there any release timeline for aioredis 2.0 ? the alpha works great so far, but using an alpha release on my main software trunk line is not the best thing to consider as an alpha version may experience major changes. Is it the documentation that delays the release?

**Comment by Andrew-Chen-Wang** [comment]
@rafaelSorel I would say ETA middle of June *at worst*, optimistically maybe beginning of May when abrookins takes over or middle of April when seandsteward returns and there isn't anything major.

TL;DR the following portion: I feel I need a second/third opinion before making the final huge leap to 2.0.0 and would need the RTD to be updated first.

Currently, it's mostly me (with finals and lots of essays coming up), so I'm taking this time to get feedback on those who are migrating or currently using it (some bugs did come up already, but they weren't major). Additionally, yes, the docs need to be updated and Idk who has power over that. I'd say the major prevention is the RTD, but that could be due to being an alpha version. Holistically speaking, there needs to be more than one person helping make major decisions and tying things together for this huge migration. (I have PTSD from PyJWT 1 to PyJWT 2 migration that broke an entire package because PyJWT 1 hadn't been updated in a 1.5 years).

**Comment by agronholm** [comment]
Hi, I was just pointed to this issue when discussing candidates for the first [AnyIO](https://github.com/agronholm/anyio) enabled redis client. Would you be willing to consider letting me base this project on it? I would pitch in with packaging and testing setup too, as I consider those my strong suits.

**Comment by abrookins** [comment]
@rafaelSorel We don't have a release timeline yet, but we'll make sure to update this issue when we do. This release needs more testing in live environments, and we also need access to the docs site.

**Comment by agronholm** [comment]
I've gone over the code and found some dangerous code: `__del__` calls code async code even though it could be triggered in any thread.

Is there a gitter room / discord server / irc channel you guys collaborate on? This would potentially go forward quicker than just posting messages here now and then.

**Comment by agronholm** [comment]
Ah, that Gitter room must be "aio-libs" which I have already joined, right?

**Comment by webknjaz** [comment]
> and we also need access to the docs site.

FWIW `latest` gets updates on pushes to `master`: https://readthedocs.org/projects/aioredis/builds/

**Comment by seandstewart** [comment]
> > and we also need access to the docs site.
> 
> FWIW `latest` gets updates on pushes to `master`: https://readthedocs.org/projects/aioredis/builds/

Yes, but if you look those docs are empty because we switched to mkdocs from Sphinx and it’s still configured for Sphinx.

**Comment by popravich** [comment]
> and we also need access to the docs site.

@seandstewart , you're now a maintainer

**Comment by mattyweb** [comment]
Is there going to be a call for testers at some point soon? I have an open source project using aioredis that's currently pinned to Python 3.7/Redis 5 but looking to update. Was wondering if there are specific stacks you're looking for feedback on.

It might be only semantics but would prefer using something with a beta label to alpha.

Stoked you're so close to release!

**Comment by seandstewart** [comment]
> > and we also need access to the docs site.
> 
> @seandstewart , you're now a maintainer

@popravich thanks for that - the new documentation is now live: https://aioredis.readthedocs.io/en/latest/

**Comment by abrookins** [comment]
> Ah, that Gitter room must be "aio-libs" which I have already joined, right?

Hi @agronholm - that Gitter room seems pretty quiet. I'm new on this project and not sure if there's a proper real-time chat somewhere, but I hang out in the Redis discord server: http://discord.gg/redis

**Comment by Vibhu-Agarwal** [comment]
Hi! Just wanted to report a few issues I faced recently.
Just to be clear, I'm a new user of aioredis and in-fact, Redis itself.
I think the issues I faced are mostly due to the inconsistencies that have arisen due to this massive change.
- First of all, I went to the docs and followed the [installation instructions](https://aioredis.readthedocs.io/en/latest/getting-started/#installation) (which installed aioredis-1.3.1)
- Then I continued through the getting-started page and copied the code-snippets in my code-editor to try them out. My IDE raised a lot of warnings and the code obviously errored out. This confused me a lot.
- I checked out the [Migrating to 2.0](https://aioredis.readthedocs.io/en/latest/migration/) section to see if there is any significant change recently. It was! But I still couldn't figure out a way to install the library without building from src. I checked out the PyPI page and the latest release was still 1.3.1. Weird.
- I googled for a few minutes and found out nothing! I thought I should just raise an issue about it and then I stumbled upon this issue and THEN I found out how to install the latest alpha release.
- I went back to check the version of which docs is serving and found out that it's still showing 1.1.3 (on the top right - beside release tag) BUT the snippets are of the latest alpha release - _the inconsistency_

I think either the docs should be switched back to the one which has guide for 1.1.3 OR the tag should be updated (along with adding the instructions on how to install the latest release) 🤷‍♂️
And btw, my new tiny project in which I'm using this alpha version: [shawarmaKoders/Hedwig](https://github.com/shawarmaKoders/Hedwig) :)

**Comment by Andrew-Chen-Wang** [comment]
@Vibhu-Agarwal Thanks for the report and good job getting it worked out. Fixed in #970

**Comment by flyser** [comment]
I tried 2.0.0a1 for a new project and it seems to work fine so far. Just a minor nitpick: the scan() function gets a "_type" parameter, which I would have expected to be called "type_" (as suggested by PEP-8).

**Comment by abrookins** [comment]
Hi folks, I wanted to post an update here about what I've been doing. Recently, I rewrote a production service to use aioredis-py with FastAPI. The migration went fairly well, and I haven't noticed any problems.

@Andrew-Chen-Wang We have docs access now IIRC. I feel pretty good about releasing 2.0 in some form. What do you think?

**Comment by Andrew-Chen-Wang** [comment]
@abrookins I agree; nothing too major with all the bug fixes so far. Even with the from_url issue, it seems like that only happens if the initial connection fails. I'm comfortable with a major release!

**Comment by agronholm** [comment]
So, trying this again: is there any interest in using structured concurrency with this library, or adding trio support? Both of which could be achieved at the same time by porting to [AnyIO](https://anyio.readthedocs.io/en/stable/)?

**Comment by abrookins** [comment]
Hi @agronholm, thanks for following up! I prefer to keep this library as basic as possible, so while I appreciate you asking, I'm not interested. @Andrew-Chen-Wang can weigh in with his thoughts if he has anything to add or feels differently.

**Comment by seandstewart** [comment]
> Hi folks, I wanted to post an update here about what I've been doing. Recently, I rewrote a production service to use aioredis-py with FastAPI. The migration went fairly well, and I haven't noticed any problems.
> 
> @Andrew-Chen-Wang We have docs access now IIRC. I feel pretty good about releasing 2.0 in some form. What do you think?
> 
> Looks like I still need to follow up on that `from_url` issue you pinged me in. I'll take a look today.

This is great news 📰.

I’m very happy to see this library is moving forward as I’ve had to step away to spend time with my family. I foresee more availability in June and look forward to helping out as I can as you take over maintenance @abrookins

**Comment by Andrew-Chen-Wang** [comment]
@agronholm I agree with abrookins. There's only so much time and effort we can put in; plus, I'm not actually that acquainted with Trio (which itself is not very mature *last I checked*). We discussed having an "extension" module, but I think the AnyIO implementation would just be a complete rewrite, so a fork is the best response I can give.

**Comment by Andrew-Chen-Wang** [comment]
> I've gone over the code and found some dangerous code: `__del__` calls code async code even though it could be triggered in any thread.
> 
> Is there a gitter room / discord server / irc channel you guys collaborate on? This would potentially go forward quicker than just posting messages here now and then.

Better late than never, but do you mind pointing to an example bit of code? I'm pretty sure all those disconnect calls always use `get_event_loop()`, but I don't think we actually support putting in your own loop anymore? And even if we do, the context is already under a particular loop so regardless it should be fine...

**Comment by abrookins** [comment]
@Andrew-Chen-Wang I've been meaning to look into this more. My test suite complains about it:

```
test_1        | tests/test_validators.py::test_skip_404_allows_non_404
test_1        |   /usr/local/lib/python3.8/asyncio/base_events.py:641: RuntimeWarning: coroutine 'Connection.disconnect' was never awaited
test_1        |   Coroutine created at (most recent call last)
test_1        |     File "/usr/local/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
test_1        |       self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
test_1        |     File "/usr/local/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
test_1        |       res = hook_impl.function(*args)
test_1        |     File "/usr/local/lib/python3.8/site-packages/_pytest/fixtures.py", line 1117, in pytest_fixture_setup
test_1        |       result = call_fixture_func(fixturefunc, request, kwargs)
test_1        |     File "/usr/local/lib/python3.8/site-packages/_pytest/fixtures.py", line 924, in call_fixture_func
test_1        |       fixture_result = fixturefunc(**kwargs)
test_1        |     File "/usr/local/lib/python3.8/site-packages/pytest_asyncio/plugin.py", line 160, in wrapper
test_1        |       return loop.run_until_complete(setup())
test_1        |     File "/usr/local/lib/python3.8/asyncio/base_events.py", line 603, in run_until_complete
test_1        |       self.run_forever()
test_1        |     File "/usr/local/lib/python3.8/asyncio/base_events.py", line 570, in run_forever
test_1        |       self._run_once()
test_1        |     File "/usr/local/lib/python3.8/asyncio/base_events.py", line 1851, in _run_once
test_1        |       handle._run()
test_1        |     File "/usr/local/lib/python3.8/asyncio/events.py", line 81, in _run
test_1        |       self._context.run(self._callback, *self._args)
test_1        |     File "/usr/local/lib/python3.8/site-packages/aioredis/connection.py", line 631, in __del__
test_1        |       coro = self.disconnect()
test_1        |     self._ready.clear()
test_1        |
test_1        | -- Docs: https://docs.pytest.org/en/stable/warnings.html
```

**Comment by Andrew-Chen-Wang** [comment]
@abrookins It may be due to `create_task`. Try replacing that call with `asyncio. run_coroutine_threadsafe(coro, loop)`

**Comment by bmerry** [comment]
I realise I'm chiming in rather late, but has any consideration been given to renaming the library rather than calling this version 2 under the same "aioredis" name? From a quick look it seems like the old API has been thrown away and replaced, and reusing the name could make migration more difficult because Python doesn't allow two versions of the same package name to coexist in a process or environment.

**Comment by Andrew-Chen-Wang** [comment]
@bmerry you can read the conversation from #891. 1.X.X hasn't been touched in two years and no one bothered to really continue its complexity. Thus, I urge you to try migrating to 2.0.0. Thanks!

**Comment by bmerry** [comment]
I have to agree with @agronholm that `Connection.__del__` looks extremely dangerous:
- It calls `asyncio.get_event_loop()`, which will return the event loop of the thread running the finalizer, which might not be the same thread that the connection was originally used with.
- Even if it is on the same thread, a finalizer can run at any time (I've been bitten by this in redis-py finalizers [before](https://github.com/jamesls/fakeredis/issues/297)), which means it's probably only safe to interact with the event loop via thread-safe idioms like `call_soon_threadsafe`.
- If the event loop is not running, it is rerun synchronously, which could block whatever code the garbage collector interrupted for some non-trivial amount of time.

In my opinion redis-py is misguided in trying to use `__del__` to close down connections. If `__del__` is getting called on open connections, then the user has chosen to let the garbage collector sort things out rather than managing the process, and so one might as well let it the garbage collector do the rest of the work too.

**Comment by bmerry** [comment]
I've spotted an issue that might cause some issues with migration from 1.x. In 1.x, you could make as many concurrent calls as you liked, and if necessary they would block until the connection pool had a free connection, even if you had a bound on the number of connections. With 2.0.0a1, trying to do too much concurrency results in an aioredis.exceptions.ConnectionError ("Too many connections").

The default in redis-py (and hence in aioredis) is 2**31 connections, which should be enough for anybody. But if someone is migrating code from 1.x with an explicit upper bound (which might be needed because 1.x defaulted to 10) they might get a nasty shock in production the first time the load is high enough to actually need all the connections and there is an exception.

I don't know what the best solution is, given the goal of following redis-py as closely as possible. Maybe it just merits a mention in the migration document that `max_connections` has this behaviour which is different to aioredis 1.3 and that it probably shouldn't be used (`aiojobs` may be an alternative to restrict the number of connections).

**Comment by erewok** [comment]
@bmerry, have you looked at using the [`BlockingConnectionPool`](https://customer.dev.mulligancloud.com/next-step-preference?rid=eax3&email=eaker@mulliganfunding.com&name=John)

**Comment by levrik** [comment]
@erewok Your posted link goes to Mulligancloud, whatever this is? Probably you had another link on the clipboard.

**Comment by bmerry** [comment]
> @bmerry, have you looked at using the [`BlockingConnectionPool`](https://customer.dev.mulligancloud.com/next-step-preference?rid=eax3&email=eaker@mulliganfunding.com&name=John)

Thanks, that's exactly the solution! (although the docstring possibly needs rewriting since it claims to be thread-safe whereas in aioredis it isn't, it is async-safe). Probably worth a mention in the migration documentation for anyone who has used a bounded connection pool in aioredis 1.x.

**Comment by erewok** [comment]
> @erewok Your posted link goes to some Mulligancloud? Probably you had another link on the clipboard.

Yeah. Bad copy pasta. Sorry about that.

**Comment by erewok** [comment]
@bmerry, can you update the link in your comment to this: https://github.com/aio-libs/aioredis-py/blob/c54fca02e216f0fb046b83e6403315fb00cd43e6/aioredis/connection.py#L1439

Not a huge deal, but could certainly confuse people in the future...

**Comment by erewok** [comment]
Is this an appropriate place to ask about the `BlockingConnectionPool` (I can start a new issue if you want)?

I have been trying to use `BlockingConnectionPool` from `v2.0.0.a1` in places where `ConnectionPool` is working fine and I keep hitting an exception on `disconnect`:

```python
  File "/location/of/dependencies/lib/python3.9/site-packages/aioredis/connection.py", line 1596, in disconnect
    async with self._lock:
AttributeError: 'BlockingConnectionPool' object has no attribute '_lock'
```

This is in this chunk of code: https://github.com/aio-libs/aioredis-py/blob/c54fca02e216f0fb046b83e6403315fb00cd43e6/aioredis/connection.py#L1598


It looks like the _direct_ reason for this is that on `__init__` a regular `ConnectionPool` ends up calling `reset`, which sets the following attributes:

```
        self._lock = asyncio.Lock()
        self._created_connections = 0
        self._available_connections = []
        self._in_use_connections = set()
```

However, the `BlockingConnectionPool` does something different in its `reset` method: https://github.com/aio-libs/aioredis-py/blob/c54fca02e216f0fb046b83e6403315fb00cd43e6/aioredis/connection.py#L1491

I am probably going to use `ConnectionPool` for now: I think the `BlockingConnectionPool` isn't totally usable yet.

**Comment by Andrew-Chen-Wang** [comment]
@erewok probably would be better as an issue now that seandstewart closed most 1.3.1 issues.

**Comment by Andrew-Chen-Wang** [comment]
@abrookins @seandstewart do you think it's worth it do add or update the CI matrix to use Redis 6.2?

**Comment by LeoQuote** [comment]
Is that possible to set timeout in redis url or through parameter of `from_url` ?

**Comment by Bautista-Baiocchi-lora** [comment]
Hello, any ETA on 2.0.0 release?

**Comment by Andrew-Chen-Wang** [comment]
@Bautista-Baiocchi-lora I want to get a second pre-release out with all our bug fixes on master + #1045 #1048 #1049 and #1051 merged by the end of this week. @seandstewart @abrookins is that possible and do you think we should instead just do a 2.0.0 release instead?

**Comment by abrookins** [comment]
@Andrew-Chen-Wang Let's get a beta release out today, do any testing we can this week, and target a Thursday release for 2.0.0.

**Comment by Andrew-Chen-Wang** [comment]
Hi all! 2.0.0b1 is now released. You can install it with `pip install --pre aioredis`! Thanks abrookins and seandstewart for keeping this going!

**Comment by abrookins** [comment]
![IMG_0302](https://user-images.githubusercontent.com/97182/125397870-64f55800-e363-11eb-828d-ed8b714bbfda.GIF)

**Comment by abrookins** [comment]
I'm holding off on the full 2.0.0 release until next week. I didn't have time to vet some of the open issues this week. I'll target spending more time on those next week, with a goal that we can release 2.0.0 by the end of the week.

**Comment by Andrew-Chen-Wang** [comment]
Note: aioredis 2.0 release will have to push back to the end of this week. I'd like to support Redis 6.2.X, and redis-py has had many awesome changes recently that we'll be merging here. Thanks for everyone's patience.

**Comment by abrookins** [comment]
@Andrew-Chen-Wang Our team is adding PRs to redis-py for 6.2.x at a blistering pace that might be hard to catch up with. What do you think about a 2.0 release this week followed closely by 6.2.x compatible commands next week? I'd like to get 2.0 out this week.

**Comment by Andrew-Chen-Wang** [comment]
@abrookins sounds good to me. A release this week at least gets people to migrate. Gettin the changes for 6.2.X can be in a minor version. For the locking mechanism, people should just follow the docs for now. Even though the tasks seems like it should just work, it's not officially supported in the docs for now. I think now's a good time to release if no more bugs have arisen.

**Comment by aviramha** [comment]
Hey, someone has mentioned before that we should have a different package name - can we consider having two at the same time? It'd ease the migration for many (including my team). We have some common infra code shared across microservices that relies on aioredis, and migrating all services at once is a bit of a mess.

**Comment by Andrew-Chen-Wang** [comment]
@aviramha That's the point of semver. Major will be breaking. Pin the package to `aioredis<2`.

**Comment by graingert** [comment]
@Andrew-Chen-Wang some people will want to use aioredis 1 and 2 in the same project

**Comment by aviramha** [comment]
@graingert Exactly. We have a library that uses redis for certain operations, and the service itself uses redis. If we update the library, all the other services need to be updated. and we can't update only the "service" redis library..

**Comment by bmoscon** [comment]
any time ANY library has a breaking change this would be an issue, please don't rename it for this. I haven't seen any other major python package change their library name when this happens

**Comment by seandstewart** [comment]
@graingert @aviramha - 

If you have a use-case where you really need to support both major versions, then you can vendor 1.X under a versioned name (e.g., aioredis-py-v1) and direct pip to install this as a separate dependency.

I have seen this done for major breaking changes where there was a need in a monolithic code-base to have a gradual rollout. I wouldn't recommend except in extreme cases, as there's an equal risk of just having two major versions of the same library and never successfully migrating, but it's a valid option.

**Comment by Andrew-Chen-Wang** [comment]
@graingert @aviramha @bmoscon We're not going to be changing the package name. People who don't know the major change and, if the 2.0 version changed its name, then no one knows about the migration process. If the 1.3.1 version package is renamed, then people's pip installation will break especially in containerized services.

What I can do rn is create a new package called aioredis-1 with package name `aioredis_1` so that you can have both. It won't be maintained, but at least it'll make your guys' teams' transition easier.

---

ah forgot vendoring was a thing

**Comment by graingert** [comment]
> @graingert @aviramha @bmoscon We're not going to be changing the package name. People who don't know the major change and, if the 2.0 version changed its name, then no one knows about the migration process. If the 1.3.1 version package is renamed, then people's pip installation will break especially in containerized services.
> 
> What I can do rn is create a new package called aioredis-1 with package name `aioredis_1` so that you can have both. It won't be maintained, but at least it'll make your guys' teams' transition easier.

Sounds great thanks!

**Comment by Andrew-Chen-Wang** [comment]
@graingert @aviramha  I like Sean's answer much better than mine. Please vendor it into your own repositories. Or fork aioredis, rename setup.py's package to whatever you want (hence renaming the package), then install from your forked git repository. Especially since there were multiple changes that we did before deciding on the migration to 2.0.0 (such as support for Python 3.8 back in November). You can fix bugs and adjust to your needs thereafter.

**Comment by abrookins** [comment]
It is time https://github.com/aio-libs/aioredis-py/pull/1075

**Comment by seandstewart** [comment]
![Hold onto your butts.](https://butt.holdings/hold-onto-your-butts.gif)

**Comment by jneines** [comment]
Huge congrats to everyone involved. I just came to study the implementation in more detail earlier the week, as I had some issues with the 1.3.1 release, just to notice that the forthcoming 2.0 will wipe all my problems away. And there it is!

This is awesome work!

**Comment by abrookins** [comment]
We've now released version 2.0.0 to PyPI. 🥳 

Thanks to @seandstewart for all your hard work on the port and @Andrew-Chen-Wang for many great PRs, issues, and replies to users. (Same goes for Sean, of course!)

Thanks also to @bmerry for many great PRs and especially insights into asyncio specifics.

And thanks to all the aioredis users who tested the alpha and beta releases to help us find bugs.

Next, we plan on adding support for new commands added to Redis since version 6.0. You can use these already by calling `execute_command()`, but helpers are better.

We will also explore ways to make some areas of the aioredis API more easily used with the asyncio `create_task()` and `gather()` APIs.

**Comment by Bautista-Baiocchi-lora** [comment]
Round of applause for the DEV team!

**Comment by bmerry** [comment]
If anyone is used to using aioredis 1.x with fakeredis, you'll be happy to know that I'm adding support for aioredis 2.0: [PR](https://github.com/jamesls/fakeredis/pull/304).

**Comment by gnat** [comment]
Wanted to drop a success story here- I've already transitioned my production over to aioredis 2.0, and have upgraded to Redis 6.

This effort by the community, @seandstewart  and @abrookins at Redis Labs saved this library. 

It's wonderful to have been a witness to this effort and I'm very happy async python continues to have a solid, maintainable, go-to Redis library. This is essential infrastructure!

**Comment by jgmarcel** [comment]
I echo @gnat’s report of success in transitioning to v2.0 and his high praise to the team of developers. Thank you, guys!

**Comment by Andrew-Chen-Wang** [comment]
Further discussion should be moved to #1225. Thank you all for the support, help, PR, and testing through these few months!
